tracer/src/Datadog.Trace/Debugger/SignatureParser.cs

```diff
@@ -0,0 +1,120 @@
+// <copyright file="SignatureParser.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Logging;
+using Datadog.Trace.VendoredMicrosoftCode.System.Buffers;
+
+#nullable enable
+namespace Datadog.Trace.Debugger
+{
+    internal static class SignatureParser
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SignatureParser));
+
+        internal static string[]? SafeParse(string? signature)
```

Comment:
nit, personally I would stick to conventions with the signature `TryParse`:

```csharp
        internal static bool SafeParse(string? signature, [NotNullWhen(true)]out string[]? signatures)

```

Reply:
No strong opinion here. Since the only callsite of this method not only don't care about it being `null`, it actually needs it to hold null then there is no special handling for the path where we fail to parse the signature.
Done in 1bd2d3aeb55726289346ca2713091b658c887517.